src/Forms/ConfirmedPasswordField.php

```diff
@@ -185,6 +248,16 @@ public function Field($properties = [])
                     $field->setAttribute($name, $value);
                 }
             }
+            if (str_ends_with($field->getName(), '[_Password]') && $this->canEvaluateStrength()) {
+                if (!$this->minPasswordStrengthExplicitlySet) {
+                    $validator = $this->getApplicableValidator();
+                    if (is_a($validator, EntropyPasswordValidator::class)) {
+                        $this->setMinPasswordStrength($validator->getRequiredStrength());
+                    }
+                }
+                $field->setAttribute('data-strength-url', $this->Link('strength'));
+                $field->setAttribute('data-min-strength', $this->getMinPasswordStrength());
```

Comment:
Why are we calling `$this->setMinPasswordStrength()`? Isn't this value only needed for the `data-min-strength` attribute?

```suggestion
                $minStrength = $this->getRequireStrongPassword() ? $this->getMinPasswordStrength() : 0;
                if ($this->getIsOnMemberForm()) {
                    $validator = Member::password_validator();
                    if (is_a($validator, EntropyPasswordValidator::class)) {
                        $minStrength = max($minStrength, $validator->getRequiredStrength());
                    }
                }
                $field->setAttribute('data-strength-url', $this->Link('strength'));
                $field->setAttribute('data-min-strength', $minStrength);
```

Set the attribute to the highest of:
1. `$this->getMinPasswordStrength()` (if `requireStrongPassword` is true)
2. The required strength of `EntropyPasswordValidator` (if we're on a member form and that's the current validator)
3. 0 (which shouldn't be used anyway because `canEvaluateStrength()` already returned true)

Reply:
A shortcut to all this would be to replace `canEvaluateStrength()` with `getMinPasswordStrengthForEvaluation()` that returns the `$minStrength` above, and returns `-1` if we shouldn't be evaluating strength in the first place. We then get both the boolean and the strength value in a single return value from a single method (which we're only calling here anyway)

So something like

```php
public function getMinPasswordStrengthForEvaluation(): int
{
    $minStrength = $this->getRequireStrongPassword() ? $this->getMinPasswordStrength() : -1;
    if ($this->getIsOnMemberForm()) {
        $validator = Member::password_validator();
        if (is_a($validator, EntropyPasswordValidator::class)) {
            $minStrength = max($minStrength, $validator->getRequiredStrength());
        }
    }
    return $minStrength;
}
```

And then this method can be updated to

```php
$minStrength = $this->getMinPasswordStrengthForEvaluation();
if ($minStrength >= 0 && str_ends_with($field->getName(), '[_Password]')) {
      $field->setAttribute('data-strength-url', $this->Link('strength'));
      $field->setAttribute('data-min-strength', $minStrength);
}
```

Reply:
Updated